### PR TITLE
Add section demonstrating JSON serialization configuration

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/12725

### CODE REVIEW

File: aspnetcore/tutorials/first-mongo-app.md [hunk 1]
```diff
@@ -203,9 +226,9 @@ The database is ready. You can start creating the ASP.NET Core web API.
 
     The preceding `BookstoreDatabaseSettings` class is used to store the *appsettings.json* file's `BookstoreDatabaseSettings` property values. The JSON and C# property names are named identically to ease the mapping process.
 
-1. Add the following code to `Startup.ConfigureServices`, before the call to `AddMvc`:
+1. Add the following highlighted code to `Startup.ConfigureServices`:
 
-    [!code-csharp[](first-mongo-app/sample/BooksApi/Startup.cs?name=snippet_ConfigureDatabaseSettings)]
+    [!code-csharp[](first-mongo-app/sample/snippets/Startup.ConfigureServices.AddDbSettings.cs?highlight=3-7)]
```
Comment (guardrex) [inline aspnetcore/tutorials/first-mongo-app.md]: This is a snapshot folder ("snippets")? If so, this is a departure from what we've (I've) been doing. My guidance might be dated tho, so please let me know if some changes were made for snapshot file location.

File: aspnetcore/tutorials/first-mongo-app.md [hunk 2]
```diff
@@ -301,6 +324,33 @@ The preceding web API controller:
     }
     ```
 
+## Configure JSON serialization options
+
+There are two details to change about the JSON responses returned in the [Test the web API](#test-the-web-api) section:
+
+* The property name camel casing should be replaced with member casing.
```
Comment (guardrex) [inline aspnetcore/tutorials/first-mongo-app.md]: "member casing" ... is that Pascal casing?

File: aspnetcore/tutorials/first-mongo-app/sample/snippets/Startup.ConfigureServices.AddDbSettings.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+{
+    services.Configure<BookstoreDatabaseSettings>(
+        Configuration.GetSection(nameof(BookstoreDatabaseSettings)));
+
+    services.AddSingleton<IBookstoreDatabaseSettings>(sp =>
+        sp.GetRequiredService<IOptions<BookstoreDatabaseSettings>>().Value);
+
+    services.AddMvc()
+            .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+}
```
Comment (guardrex) [inline aspnetcore/tutorials/first-mongo-app/sample/snippets/Startup.ConfigureServices.AddDbSettings.cs]: Add a line feed to make the 🙏 *GH server gods* :pray: happy. :smile:

File: aspnetcore/tutorials/first-mongo-app/sample/snippets/Startup.ConfigureServices.AddSingletonService.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        Configuration.GetSection(nameof(BookstoreDatabaseSettings)));
+
+    services.AddSingleton<IBookstoreDatabaseSettings>(sp =>
+        sp.GetRequiredService<IOptions<BookstoreDatabaseSettings>>().Value);
+
+    services.AddSingleton<BookService>();
+
+    services.AddMvc()
+            .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+}
```
Comment (guardrex) [inline aspnetcore/tutorials/first-mongo-app/sample/snippets/Startup.ConfigureServices.AddSingletonService.cs]: Add a line feed.

File: aspnetcore/tutorials/first-mongo-app.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/tutorials/first-mongo-app.md]: I'll follow what you've done in the past.

Comment (guardrex) [reply aspnetcore/tutorials/first-mongo-app.md]: Almost. lol 

Like this with the version inside the folder.

https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/host/generic-host

Only the `_snapshot` has to be added. The files are stored in parallel generally. Well ... I'm trying. This has been an evolving (for the better I hope) thing. This current 2.2 round of sample updates on the tracking issue I'm trying to bring it all together.

**Pull request merged**